File: backend-JS/src/controllers/authController.js
const { authService } = require('../services');
const { success, created, error } = require('../utils/response');

const register = async (req, res) => {
  try {
    const result = await authService.register(req.body);
    return created(res, result, '注册成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const login = async (req, res) => {
  try {
    const clientIP = req.ip || req.connection.remoteAddress;
    const result = await authService.login(req.body, clientIP);
    return success(res, result, '登录成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const logout = async (req, res) => {
  try {
    const clientIP = req.ip || req.connection.remoteAddress;
    await authService.logout(req.userId, clientIP);
    return success(res, null, '登出成功');
  } catch (err) {
    return error(res, 500, err.message);
  }
};

const getMe = async (req, res) => {
  try {
    const user = await authService.getCurrentUser(req.userId);
    return success(res, { user }, '获取成功');
  } catch (err) {
    return error(res, 404, err.message);
  }
};

const getProfile = async (req, res) => {
  try {
    const profile = await authService.getUserProfile(req.userId);
    return success(res, { user: profile }, '获取成功');
  } catch (err) {
    return error(res, 404, err.message);
  }
};

const changePassword = async (req, res) => {
  try {
    const clientIP = req.ip || req.connection.remoteAddress;
    await authService.changePassword(req.userId, req.body, clientIP);
    return success(res, null, '密码修改成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const refreshToken = async (req, res) => {
  try {
    const result = await authService.refreshToken(req.body.token);
    return success(res, result, '令牌刷新成功');
  } catch (err) {
    return error(res, 401, err.message);
  }
};

module.exports = {
  register,
  login,
  logout,
  getMe,
  getProfile,
  changePassword,
  refreshToken
};

File: backend-C#/Controllers/AuthController.cs
using IMS.DTOs;
using IMS.DTOs.Requests;
using IMS.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace IMS.Controllers;

[ApiController]
[Route("api/auth")]
public class AuthController : BaseController
{
    private readonly IAuthService _authService;

    public AuthController(IAuthService authService)
    {
        _authService = authService;
    }

    [HttpPost("register")]
    [AllowAnonymous]
    public async Task<ActionResult<ApiResponse>> Register([FromBody] RegisterRequest request)
    {
        try
        {
            var clientIp = HttpContext.Connection.RemoteIpAddress?.ToString() ?? "unknown";
            var result = await _authService.RegisterAsync(request, clientIp);
            return Created(new { user = result.User, token = result.Token }, "注册成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("login")]
    [AllowAnonymous]
    public async Task<ActionResult<ApiResponse>> Login([FromBody] LoginRequest request)
    {
        try
        {
            var clientIp = HttpContext.Connection.RemoteIpAddress?.ToString() ?? "unknown";
            var result = await _authService.LoginAsync(request, clientIp);
            return Success(new { user = result.User, token = result.Token }, "登录成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("logout")]
    [Authorize]
    public async Task<ActionResult<ApiResponse>> Logout()
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        var clientIp = HttpContext.Connection.RemoteIpAddress?.ToString() ?? "unknown";
        await _authService.LogoutAsync(CurrentUserId.Value, clientIp);
        return Success("登出成功");
    }

    [HttpGet("me")]
    [Authorize]
    public async Task<ActionResult<ApiResponse>> GetMe()
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        var user = await _authService.GetCurrentUserAsync(CurrentUserId.Value);
        if (user == null)
        {
            return Error("用户不存在", 404);
        }

        return Success(new { user }, "获取成功");
    }

    [HttpGet("profile")]
    [Authorize]
    public async Task<ActionResult<ApiResponse>> GetProfile()
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        var profile = await _authService.GetUserProfileAsync(CurrentUserId.Value);
        if (profile == null)
        {
            return Error("用户不存在", 404);
        }

        return Success(new { user = profile }, "获取成功");
    }

    [HttpPost("change-password")]
    [Authorize]
    public async Task<ActionResult<ApiResponse>> ChangePassword([FromBody] ChangePasswordRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var clientIp = HttpContext.Connection.RemoteIpAddress?.ToString() ?? "unknown";
            await _authService.ChangePasswordAsync(CurrentUserId.Value, request, clientIp);
            return Success("密码修改成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("refresh")]
    [AllowAnonymous]
    public async Task<ActionResult<ApiResponse>> RefreshToken([FromBody] RefreshTokenRequest request)
    {
        try
        {
            var result = await _authService.RefreshTokenAsync(request);
            if (result == null)
            {
                return Error("刷新令牌失败", 401);
            }
            return Success(new { user = result.User, token = result.Token }, "令牌刷新成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 401);
        }
    }
}
